### PR TITLE
CC_SLURM_NHC (Support killing NHC via SLURM Prolog)

### PR DESCRIPTION
**Pull request by garvct**
-Added support to kill NHC via SLURM Prolog

### CODE REVIEW

File: experimental/cc_slurm_nhc/cc_slurm_nhc/specs/default/cluster-init/files/configure_nhc.sh [hunk 1]
```diff
@@ -87,6 +88,13 @@ function slurm_config() {
       echo "HealthCheckProgram=${NHC_EXE}" >> $SLURM_CONF
       echo "HealthCheckInterval=${SLURM_HEALTH_CHECK_INTERVAL}" >> $SLURM_CONF
       echo "HealthCheckNodeState=${SLURM_HEALTH_CHECK_NODE_STATE}" >> $SLURM_CONF
+      grep -qi '^Prolog' $SLURM_CONF
```
Comment (vanzod) [inline experimental/cc_slurm_nhc/cc_slurm_nhc/specs/default/cluster-init/files/configure_nhc.sh:91]: If another project installed a prolog script (e.g. Pyxis/Enroot) the NHC kill will never be set in place.
My suggestion would be to install the `kill_nhc.sh` script in `/sched/scripts` and execute such script in a common prolog file (e.g. /sched/scripts/prolog.sh) which is then executed by slurmd.

Comment (garvct) [reply experimental/cc_slurm_nhc/cc_slurm_nhc/specs/default/cluster-init/files/configure_nhc.sh:91]: Standardized slurm epilog and prolog amongst different cyclecloud projects, see configure_nhc.sh

File: experimental/cc_slurm_nhc/cc_slurm_nhc/specs/default/cluster-init/files/configure_nhc.sh [hunk 2]
```diff
@@ -78,6 +79,28 @@ function nhc_sysconfig() {
 }
 
 
+function update_slurm_prolog_epilog() {
+
+   prolog_epilog=$1
+   script=$2
+   grep -qi /sched/scripts/${prolog_epilog}.sh $SLURM_CONF
+   prolog_epilog_does_not_exist=$?
+   cp $CYCLECLOUD_SPEC_PATH/files/$script /sched/scripts
```
Comment (vanzod) [inline experimental/cc_slurm_nhc/cc_slurm_nhc/specs/default/cluster-init/files/configure_nhc.sh:88]: Add a chmod to ensure the script is executable

File: experimental/cc_slurm_nhc/cc_slurm_nhc/specs/default/cluster-init/files/configure_nhc.sh [hunk 3]
```diff
@@ -78,6 +79,28 @@ function nhc_sysconfig() {
...
+   if [[ $prolog_does_not_exist == 1 ]]; then
+      if [[ $prolog_epilog == "prolog" ]]; then
+         cp $CYCLECLOUD_SPEC_PATH/files/prolog.sh /sched/scripts
+         echo "Prolog=/sched/prolog.sh" >> $SLURM_CONF
+         echo "PrologFlags=Alloc" >> $SLURM_CONF
+      elif [[ $prolog_epilog == "epilog" ]]; then
+         cp $CYCLECLOUD_SPEC_PATH/files/epilog.sh /sched/scripts
```
Comment (vanzod) [inline experimental/cc_slurm_nhc/cc_slurm_nhc/specs/default/cluster-init/files/configure_nhc.sh:95]: Chmod here too?

Comment (garvct) [reply experimental/cc_slurm_nhc/cc_slurm_nhc/specs/default/cluster-init/files/configure_nhc.sh:97]: All done

**Review by vanzod — APPROVED** [review]

**Pull request merged**